### PR TITLE
Allow Galaxy database to be ahead of the codebase.

### PR DESCRIPTION
**Pull request by jmchilton**
Helps development for sure and there are situations when it would help production-y setups - if you get some migrations in a release branch you don't want and need to backup later after the update but don't want to loose the data you've added in the meantime. For most migrations this would be fine it would be better to loose the data - for some though this would be problematic and Galaxy wouldn't work. I'd merge this - but I understand if we don't want to.

### CODE REVIEW

File: lib/galaxy/model/migrate/check.py [hunk 1]
```diff
@@ -89,14 +89,21 @@ def migrate():
...
     db_schema = schema.ControlledSchema(engine, migrate_repository)
-    if migrate_repository.versions.latest != db_schema.version:
+    config_arg = ''
+    if galaxy_config_file and os.path.abspath(os.path.join(os.getcwd(), 'config', 'galaxy.ini')) != galaxy_config_file:
+        config_arg = ' -c %s' % galaxy_config_file.replace(os.path.abspath(os.getcwd()), '.')
+    template_args = (db_schema.version, migrate_repository.versions.latest, config_arg)
+    if migrate_repository.versions.latest > db_schema.version:
         config_arg = ''
         if galaxy_config_file and os.path.abspath(os.path.join(os.getcwd(), 'config', 'galaxy.ini')) != galaxy_config_file:
             config_arg = ' -c %s' % galaxy_config_file.replace(os.path.abspath(os.getcwd()), '.')
```
Comment (nsoranzo) [inline lib/galaxy/model/migrate/check.py]: It seems to me you can now remove these last 3 lines, right?

**Comment by martenson** [comment]
Is the main goal for this PR to help with development? To me this does not feel like a big improvement and has potential drawbacks for deployments. What are the best dev usecases for this?

**Comment by hexylena** [comment]
Literally any time I have to switch from working on a branch to testing someone's older, existing PR that branched off of dev months ago. Then I've already checked out the target revision, start up galaxy "oh, no, sorry, you need 132". Of course because I'm back in time, migration files 133-135 don't exist so I have to check out dev / wherever I was, downgrade, go back to what I was doing. I personally find this rather annoying. This happens at least once or twice per month to me.

**Comment by martenson** [comment]
@erasche I usually only change the value in the DB so it thinks to be correct (which I guess is almost what this PR does :D)

**Comment by jmchilton** [comment]
I guess I will reopen and make it contingent on database_auto_migrate - hopefully if that is set we are okay with running ahead because it is likely a dev database.

**Comment by martenson** [comment]
@jmchilton that sounds like a good compromise!

**Comment by hexylena** [comment]
but this could be used in prod for blue/green deployments? Users could run a galaxy-test instances off of the same data+database as galaxy-main which would be nice for testing workflows / etc without having to copy the data between the test/main.

**Comment by martenson** [comment]
@erasche This feels dirty to me, but I understand the use case for some. I am +1 on this if it is tied to `database_auto_migrate` config option. Wanna reopen @jmchilton ?

**Pull request closed without merging**